packages/vite/src/node/plugins/html.ts

```diff
@@ -222,7 +227,11 @@ function handleParseError(
  * Compiles index.html into an entry js module
  */
 export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
-  const [preHooks, postHooks] = resolveHtmlTransforms(config.plugins)
+  const [preHooks, postHooks] = resolveHtmlTransforms([
+    preImportMapHook(config),
+    ...config.plugins,
+    postImportMapHook()
```

Comment:
It seems hooks are not working for build because `resolveHtmlTransforms` receives a plugin, not a hook itself.

Reply:
omg thanks for catching this. Not sure what's wrong with me but not my proudest PR so far 😅

Reply:
Aight I manually insert the hook into `preHooks` and `postHooks` for now. That should remove the need to destructure on every transform. I'm thinking leaving as hooks would be nicer for now instead of a plugin so Vite plugins are only passed around in `plugins/index.ts`.